proto/schema.proto

```diff
@@ -0,0 +1,170 @@
+  // This field is REQUIRED.
+  uint64 byte_offset = 1;
+
+  // This is the size of the data range.
+  // size_bytes MUST be greater than zero.
+  // This field is REQUIRED.
+  uint64 size_bytes = 2;
+}
+
+// The GetMetadataAllocatedRequest message is used to solicit metadata
+// on the allocated blocks of a snapshot: i.e. this identifies the
+// data ranges that have valid data as they were the target of some
+// previous write operation on the volume.
+message GetMetadataAllocatedRequest {
+  // This field specifies an audience scoped authentication token presented by client.
+  // This token is used to authenticate and authorize the request at server side.
+  // This field is REQUIRED.
+  string security_token = 1;
+
+  // This field indicates the Kubernetes namespace where the VolumeSnapshot exists.
+  // If not set, the VolumeSnapshot will be looked for in the "default" namespace.
+  // This field is OPTIONAL.
+  string namespace = 2;
+
+  // This is the VolumeSnapshot object name against which changes
+  // are to be computed.
+  // This field is REQUIRED.
+  string snapshot_name = 3;
+
+  // This indicates the zero based starting byte position in the volume
+  // snapshot from which the result should be computed.
+  // It is intended to be used to continue a previously interrupted
+  // call.
+  // The client SHOULD specify this value to be the offset of the byte
+  // position immediately after the last byte of the last data range
+  // received, if continuing an interrupted operation, or zero if not.
+  // This field is REQUIRED.
+  uint64 starting_offset = 4;
+
+  // This is an optional parameter, and if non-zero it specifies the
+  // maximum number of tuples to be returned in each
+  // GetMetadataAllocatedResponse message returned by the RPC stream.
+  // The plugin will determine an appropriate value if 0, and is
+  // always free to send less than the requested value.
+  // This field is OPTIONAL.
+  uint32 max_results = 5;
+}
+
+// GetMetadataAllocatedResponse messages are returned in gRPC stream.
+// Cumulatively, they provide information on the allocated data
+// ranges in the snapshot.
+message GetMetadataAllocatedResponse {
+  // This specifies the style used in the BlockMetadata sequence.
+  // This value must be the same in all such messages returned by
+  // the stream.
+  // If block_metadata_type is FIXED_LENGTH, then the size_bytes field
+  // of each message in the block_metadata list MUST be constant.
+  // This field is REQUIRED.
+  BlockMetadataType block_metadata_type = 1;
+
+  // This returns the size of the underlying volume.
+  // This value must be the same in all such messages returned by
+  // the stream.
+  // This field is REQUIRED.
+  uint64 volume_size_bytes = 2;
+
+  // This is a list of data range tuples.
+  // If the value of max_results in the GetMetadataDeltaRequest message
+  // is greater than zero, then the number of entries in this list MUST
+  // be less than or equal to that value.
```

Comment:
Again, the CSI spec comment is more explicit:

```
  // This is a list of data range tuples.
  // If the value of max_results in the GetMetadataAllocatedRequest
  // message is greater than zero, then the number of entries in this
  // list MUST be less than or equal to that value.
  // The SP MUST respect the value of starting_offset in the request.
  // The byte_offset fields of adjacent BlockMetadata messages
  // MUST be strictly increasing and messages MUST NOT overlap:
  // i.e. for any two BlockMetadata messages, A and B, if A is returned
  // before B, then (A.byte_offset + A.size_bytes <= B.byte_offset)
  // MUST be true.
  // This MUST also be true if A and B are from block_metadata lists in
  // different GetMetadataAllocatedResponse messages in the gRPC stream.
```
We should change `SP` to `sidecar` and `MUST` to `will`.